### DISCUSSION TITLE
First proof-of-concept landlock jail exists

### DISCUSSION THREAD

**Discussion by osevan** [netblue30/firejail#5130] [Show and tell]
Can anyone tell if someone already working firejail with landlocked userland support? 

Here is first jail available :

https://github.com/gnoack/landlockjail/blob/main/lljail.c
+
Thanks and

Best regards

**Comment by rusty-snake** [comment]
No.

What do you expect from landlock support? Why/What is better than the current filesystem restrictions? Can you explain what you want and why in a more detailed way than "There's a new kernel features, please use it".

**Comment by rusty-snake** [comment]
> https://github.com/gnoack/landlockjail/blob/main/lljail.c

If you want to play with landlock and want more control than `ACCESS_FILE`, `ACCESS_FS_ROUGHLY_READ` and `ACCESS_FS_ROUGHLY_WRITE` (e.g. `ACCESS_FS_ROUGHLY_READ & ~LANDLOCK_ACCESS_FS_EXECUTE`): https://codeberg.org/crabjail/crablock
